src/main/java/org/apache/commons/lang3/ObjectUtils.java

```diff
@@ -215,6 +216,199 @@ public static boolean anyNull(final Object... values) {
         return !allNotNull(values);
     }
 
+    /**
+     * Applies a function to a value if it's not {@code null}. The
+     * function is only applied if the value is not {@code null},
+     * otherwise the method returns {@code null} immediately. If the
+     * value is not {@code null} then the result of the function is
+     * returned.
+     *
+     * <pre>
+     * ObjectUtils.applyIfNotNull("a", String::toUpperCase)  = "A"
+     * ObjectUtils.applyIfNotNull(null, String::toUpperCase) = null
+     * ObjectUtils.applyIfNotNull("a", s -&gt; null)         = null
+     * </pre>
+     *
+     * Useful when working with expressions that may return {@code null}
+     * as it allows a single-line expression without making temporary
+     * local variables or evaluating expressions twice. Provides an
+     * alternative to using {@link Optional} that is shorter and has
+     * less allocation.
+     *
+     * <pre>
+     * String name = applyIfNotNull(peopleMap.get(key), Person::getName);
+     *
+     * // Alternative - requires local to avoid calling Map.get twice
+     * Person person = peopleMap.get(key);
+     * String name = (person != null) ? person.getName() : null;
+     *
+     * // Alternative with Optional - idiomatic, but longer and requires
+     * // allocation
+     * String name = Optional.ofNullable(peopleMap.get(key))
+     *                       .map(Person::getName)
+     *                       .orElse(null);
+     * </pre>
+     *
+     * @param <T> The type of the input value.
+     * @param <R> The type of the returned value.
+     * @param value The value to apply the function to, may be
+     * {@code null}.
+     * @param mapper The function to apply, must not be {@code null}.
+     * @return The result of the function (which may be {@code null})
+     * or {@code null} if the input value is {@code null}.
+     * @since 3.19.0
+     */
+    public static <T, R> R applyIfNotNull(
+            final T value,
```

Comment:
I didn't add `@Nullable` for these methods because they're not in other methods in this class, but let me know if you want them. It might need some changes in the project build dependencies though?